### DISCUSSION TITLE
Is it possible to pass flags to envsubst?

### DISCUSSION THREAD

**Discussion by embee1981** [mikefarah/yq#1136] [Q&A] [answered]
Hi all

I'm looking to use the -no-unset flag that envsubst supports at https://github.com/a8m/envsubst#imposing-restrictions

Is this possible, something like?

 yq '(.. | select(tag == "!!str")) |= envsubst -no-unset' configFile.yaml

I've tried --no-unset, -no-unset and no-unset after the envsubst but its not happy with any of these. Just checking if I'm invoking wrong or this is something that's not supported.

I'm playing with files that use environment variables and it would be handy to use this feature since its a nifty enhancement over what's available via the standard command line

**Comment by mikefarah** [comment]
Nah its not yet supported - moving this to an issue to track

**Reply by embee1981** [reply]
Ahh oki. Thanks Mike

**Comment by ngirard** [comment, marked as answer]
I think it's possible now, via
```sh
yq '(.. | select(tag == "!!str")) |= envsubst(nu) configFile.yaml
```

**Reply by mikefarah** [reply]
Yep - you are right! forgot to update this discussion, see https://mikefarah.gitbook.io/yq/operators/env-variable-operators#envsubst-options for all the options :)
